frame/name-service/src/commit_reveal.rs

```diff
@@ -0,0 +1,183 @@
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Handles the commit and reveal part of the registration process.
+
+use crate::{types::*, *};
+use frame_support::{
+	pallet_prelude::*,
+	traits::{Currency, ExistenceRequirement, OnUnbalanced, ReservableCurrency, WithdrawReasons},
+};
+use sp_runtime::traits::{Saturating, Zero};
+use sp_std::prelude::*;
+
+impl<T: Config> Pallet<T> {
+	/// Get the commitment hash from the raw name and secret.
+	pub fn commitment_hash(name: &[u8], secret: u64) -> CommitmentHash {
+		sp_io::hashing::blake2_256(&(name, secret).encode())
+	}
+
+	/// Get the name hash from raw bytes.
+	pub fn name_hash(name: &[u8]) -> NameHash {
+		sp_io::hashing::blake2_256(name)
+	}
+
+	/// Returns a commitment by hash if it exists.
+	pub fn get_commitment(
+		commitment_hash: CommitmentHash,
+	) -> Result<CommitmentOf<T>, DispatchError> {
+		Commitments::<T>::get(commitment_hash).ok_or(Error::<T>::CommitmentNotFound.into())
+	}
+
+	/// Checks whether a commitment has passed the minimum commitment period.
+	pub fn is_commitment_valid(
+		commitment: &CommitmentOf<T>,
+		block_number: &T::BlockNumber,
+	) -> bool {
+		&commitment.when.saturating_add(T::MinCommitmentAge::get()) < block_number
+	}
+
+	/// Checks whether a commitment has passed the commitment expiry time.
+	pub fn is_commitment_expired(
+		commitment: &CommitmentOf<T>,
+		block_number: &T::BlockNumber,
+	) -> bool {
+		&commitment.when.saturating_add(T::MaxCommitmentAge::get()) < block_number
+	}
+
+	pub fn do_commit(
+		depositor: T::AccountId,
+		owner: T::AccountId,
+		commitment_hash: CommitmentHash,
+	) -> DispatchResult {
+		ensure!(!Commitments::<T>::contains_key(commitment_hash), Error::<T>::CommitmentExists);
+
+		let block_number = frame_system::Pallet::<T>::block_number();
+		let deposit = T::CommitmentDeposit::get();
+
+		T::Currency::reserve(&depositor, deposit)?;
+
+		let commitment = Commitment {
+			owner: owner.clone(),
+			when: block_number,
+			depositor: depositor.clone(),
+			deposit,
+		};
+
+		Commitments::<T>::insert(commitment_hash, commitment);
+		Self::deposit_event(Event::<T>::Committed { depositor, owner, hash: commitment_hash });
+		Ok(())
+	}
+
+	pub fn do_reveal(
+		fee_payer: T::AccountId,
+		name: Vec<u8>,
+		secret: u64,
+		length: T::BlockNumber,
+	) -> DispatchResult {
+		ensure!(name.len() <= T::MaxNameLength::get() as usize, Error::<T>::NameTooLong);
```

Comment:
Also, instead of preventing names that are too short by setting an unresonably high registration fee, why can't the check be performed here?

Reply:
I think casting a `u32` to a `usize` is not a problem, but I can still change it.

> Also, instead of preventing names that are too short by setting an unresonably high registration fee, why can't the check be performed here?  

Yea that should be added.

Reply:
i think casting to usize is fine. this behavior exists a lot of places afaik